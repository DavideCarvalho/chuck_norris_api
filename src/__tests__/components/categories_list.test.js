import shallowWithStore from '../../shallowWithStore';
import Enzyme, { shallow } from 'enzyme';
import CategoriesList from '../../components/categories_list';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Categories List Component', () => {
  it('should render', () => {
    const component = shallow(<CategoriesList fetchJokesCategories={() => null} />);
    expect(component).toMatchSnapshot();
  })
})