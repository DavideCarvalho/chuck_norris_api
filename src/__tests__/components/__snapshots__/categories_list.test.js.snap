// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories List Component should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CategoriesList
    fetchJokesCategories={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Col
          span={6}
>
          Category 1
</Col>,
        <Col
          span={6}
>
          Category 2
</Col>,
        <Col
          span={6}
>
          Category 3
</Col>,
        <Col
          span={6}
>
          Category 4
</Col>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Category 1",
          "span": 6,
        },
        "ref": null,
        "rendered": "Category 1",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Category 2",
          "span": 6,
        },
        "ref": null,
        "rendered": "Category 2",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Category 3",
          "span": 6,
        },
        "ref": null,
        "rendered": "Category 3",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Category 4",
          "span": 6,
        },
        "ref": null,
        "rendered": "Category 4",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Col
            span={6}
>
            Category 1
</Col>,
          <Col
            span={6}
>
            Category 2
</Col>,
          <Col
            span={6}
>
            Category 3
</Col>,
          <Col
            span={6}
>
            Category 4
</Col>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Category 1",
            "span": 6,
          },
          "ref": null,
          "rendered": "Category 1",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Category 2",
            "span": 6,
          },
          "ref": null,
          "rendered": "Category 2",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Category 3",
            "span": 6,
          },
          "ref": null,
          "rendered": "Category 3",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Category 4",
            "span": 6,
          },
          "ref": null,
          "rendered": "Category 4",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
